core/src/main/scala/com/lightbend/paradox/markdown/Snippet.scala

```diff
@@ -28,16 +26,40 @@ object Snippet {
   class SnippetException(message: String) extends RuntimeException(message)
 
   def apply(file: File, labels: Seq[String]): (String, String) = {
-    (extract(file, labels), language(file))
+    val lines = Source.fromFile(file)("UTF-8").getLines.toSeq
```

Comment:
this leaks file descriptor doesnt it?
gotta close the source

Reply:
Good point, the lines used to be read multiple times like this...

Reply:
There's a few spots below as well, please skim for the leak :)